### ISSUE TITLE
Is injection possible even when BM is not running?

### ISSUE THREAD

**Issue by binury** [bakkesmodorg/BakkesModSDK#14]
I accidentally encountered a scenario recently where -I suspect- installation of a plugin caused client instability, even once disabled.

I was checking out new plugins through the web app, installed a few. Later, the game began to crash 100% under specific circumstances (joining an unranked match via Steam friends list when one of the players was using an octane with a specific item equipped lol?). I unloaded BM, deleted RL caches, reinstalled RL totally, updated drivers. Nothing helped.
The only clue was seeing BM with my config and extra plugins working after re-installation- so I knew that BM's config had persisted. mving it and then reinstalling BM resolved the issue completely. **That meant something in the `bakkesmod` folder was causing the game to crash even when the client was uncooked.**

TL;DR:
Not asking for any help- my issue is resolved but I asked before in Discord but to no avail-  how is it possible for plugins (assumed) to impact client stability even even when BM is not running? Is this unusual or actually expected by design?

P.S.: If it is unusual or unexpected I still have a copy of the `bakkesmod` folder I can share to replicate the problem.

**Comment by binury** [comment]
[bakkesmod.zip](https://github.com/bakkesmodorg/BakkesModSDK/files/4640609/bakkesmod.zip)

**Comment by Bakkes** [comment]
No, injection should not be possible if BM isn't running. The way injection is done and plugins are loaded is:
- The bm injector checks if the RL process is running
- If it is, it will inject the bakkesmod.dll
- Once the bakkesmod.dll is loaded and initialized inside of RL's memory, it will load the plugins defined in plugins.cfg

No Rocket League game files on disk are altered at all by BM and plugins currently on the BakkesPlugins website. The only way of how the Rocket League executable itself could be "broken" is if anti virus software would interfere in some way

Hate to ask, but are you sure the injector was actually fully closed and not just hidden to the system train, while being removed from the taskbar?

There are .dmp files in My Documents/My Games/Rocket League/TAGame/Logs, if you upload those here we could maybe identify which plugin was causing the issue

**Comment by binury** [comment]
>  are you sure the injector was actually fully closed

Positive BM was not running.

I wasn't able too look at this too much, sorry- Visual Studio community, despite being free, requires authentication.
[RL-v200401.53214.269849-2020.05.15-21.32.49.zip](https://github.com/bakkesmodorg/BakkesModSDK/files/4647161/RL-v200401.53214.269849-2020.05.15-21.32.49.zip)

**Comment by Bakkes** [comment]
This dump shows that BM was definitely running and not a plugin loading by itself though. The bakkesmod DLL was definitely injected somehow (and the first BM related DLL that was loaded), and I can't think of a way how that would happen if the bakkesmod injector (or branksmod version) wasn't running.

Anyway, the crash seems to be caused by the dejavu plugin, so @adamk33n3r might want to take a look at the crashdump, but for now I'm closing this issue.

Thanks for your report though, if you do run into the same issue again maybe a memory dump/dump of listed processes might help, but for now im unsure how to proceed with this issue.

**Comment by binury** [comment]
Sorry, that is the one and only dmp I have since reinstalling. I’ll try and replicate this to generate another, without the injector loaded.
